Add the ruff linter and formatter to recceiver

jacomago opened the pull request:
Adds the ruff linter and formatter https://docs.astral.sh/ruff/

Checks for problematic code and auto formats code.

Removes unused code

jacomago committed:
Add ruff to pre-commit

jacomago committed:
Update readmes

jacomago committed:
Add basic ruff instructions

jacomago committed:
Auto ruff fixes

jacomago committed:
Auto ruff formatting

jacomago committed:
Fix F841 ruff lint

jacomago committed:
Fix E402 ruff lint

jacomago committed:
fix v2 print statements

jacomago committed:
fix test_client problems

jacomago committed:
fix test files

jacomago committed:
Fix E722 ruff lint

jacomago committed:
Use correct permissions on reading file

simon-ess commented on server/recceiver/test_cfstore.py:
```diff
@@ -29,7 +29,7 @@ def test_3_iocs(self):
             7: ("ch7", "longout"),
         }
-        deferred = yield self.cfstore.commit(TR1)
+        _ = yield self.cfstore.commit(TR1)
```
I'm a bit surprised by this. Wouldn't it be better to just do
```py
yield self.cfstore.commit(TR1)
```
?

jacomago replied:
Yes. Probably. Tempted to just remove this test file entirely as it is sooo out of date and incompatible with other cfstore changes.

simon-ess commented on test-client.py:
```diff
@@ -2,12 +2,12 @@
 S.bind(('',5049))
 
-print S, S.fileno()
+print(S, S.fileno())
```
Wait, is this python2 or python3 code? Were these tests being run?

jacomago replied:
I'm guessing python2 code. I think this is more code to create a quick and dirty python reccaster. Could just remove?

jacomago committed:
Remove old unused test

jacomago committed:
fix open file mode

simon-ess commented on server/recceiver/application.py:
```diff
@@ -79,68 +81,75 @@ def privilegedStartService(self):
-                 # newer Twisted errors. sigh...
+            pass  # older Twisted required this.
+            # newer Twisted errors. sigh...
 
```
This comment is weirdly aligned.

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -187,94 +204,138 @@ def _commitWithThread(self, TR):
             pvInfo[rid] = {"pvName": rname}
-            if (self.conf.get('recordType', 'default' == 'on')):
-                pvInfo[rid]['recordType'] = rtype
+            if self.conf.get("recordType", "default" == "on"):
```
This  looks... weird. Should this be instead
```py
if self.conf.get("recordType", default="on"):
```
Otherwise, this is just equivalent to
```py
if self.conf.get("recordType", False):
```
???

jacomago replied:
Fixed by swapping to just if self.conf.get("recordType")

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -187,94 +204,138 @@ def _commitWithThread(self, TR):
             self.iocs[iocid]["channelcount"] += 1
             """In case, alias exists"""
-            if (self.conf.get('alias', 'default' == 'on')):
+            if self.conf.get("alias", "default" == "on"):
```
Here too. `"default"` is never equal to `"on"`.

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -187,94 +204,138 @@ def _commitWithThread(self, TR):
-                        self.channel_dict[a].append(iocid)  # add iocname to pvName in dict
+                        self.channel_dict[a].append(
+                            iocid
+                        )  # add iocname to pvName in dict
```
This comment is both a bit superfluous and slightly incorrect.

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -187,94 +204,138 @@ def _commitWithThread(self, TR):
                 self.remove_channel(pv, iocid)
                 """In case, alias exists"""
-                if (self.conf.get('alias', 'default' == 'on')):
+                if self.conf.get("alias", "default" == "on"):
```
Another one.

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -356,96 +441,216 @@ def __updateCF__(proc, pvInfoByName, delrec, hostName, iocName, iocIP, iocid, ow
+                        ),
+                        ch["properties"],
+                    )
+                    if conf.get("recordType", "default") == "on":
```
Ooooooh maybe this is what it should have been in those earlier examples.

simon-ess commented on server/recceiver/cfstore.py:
```diff
@@ -567,26 +803,42 @@ def getCurrentTime(timezone=False):
+                    retry_seconds=retry_seconds
+                )
+            )
+            # _log.debug(str(channels_dict))
```
Commented code

simon-ess commented on server/recceiver/recast.py:
```diff
@@ -133,25 +137,29 @@ def recvClientGreeting(self, body):
 
     # 0x0002
     def recvPong(self, body):
-        nonce, = _ping.unpack(body[:_ping.size])
```
This looks odd to me, why is it not just
```py
nonce = _ping.unpack(...)
```
Or is the point that it is a list with a single element? I guess then
```py
nonce, *_ = _ping.unpack(...)
```
but maybe this doesn't matter...

jacomago replied:
I think unpack returns a tuple (can actually see in the docs)

anderslindho replied:
I though the convention was to prefer `foo,` over `(foo,)`

jacomago committed:
Badly aligned comment fix

jacomago committed:
Reading cfstore conf "on" variables

Pretty clear this has been broken for the "alias" and "recordType" settings for a long time.
I've set both options now that if they are set, then they are on. I expect most people turn both options on.

anderslindho reviewed (approved):
LGTM
